src/execution/interpreter_loop.rs

```diff
@@ -15,22 +15,85 @@ use alloc::vec::Vec;
 use crate::{
     assert_validated::UnwrapValidatedExt,
     core::{
-        indices::{FuncIdx, GlobalIdx, LocalIdx},
+        indices::{DataIdx, FuncIdx, GlobalIdx, LocalIdx},
         reader::{
             types::{memarg::MemArg, FuncType},
             WasmReadable, WasmReader,
         },
     },
     locals::Locals,
-    store::Store,
+    store::{DataInst, Store},
     value,
     value_stack::Stack,
-    NumType, RuntimeError, ValType, Value,
+    Limits, NumType, RuntimeError, ValType, Value,
 };
 
 #[cfg(feature = "hooks")]
 use crate::execution::hooks::HookSet;
 
+/// Returns a tuple referring to -> (relative_address, data_to_store)
```

Comment:
I have extracted part of functionality of the `i32.store8` and `i32.load8_u` instructions here. This is because they are reused in other instructions related to memory and didn't want to pop and push off the stack every time (efficiency related problem).